### ISSUE TITLE
Microsoft Entra ID connector - wrong scope in body

### ISSUE THREAD

**Issue by morpulse** [airbytehq/airbyte#47870]
### Connector Name

source-microsoft-entra-id

### Connector Version

0.0.2

### What step the error happened?

None

### Relevant information

Trying to configure the connector via the UI results in 400 Bad Request error. After digging in using the Builder, I noticed the 'refresh_request_body' is set to:
scope: {{ config['application_id_uri'] }}

while it should be https://graph.microsoft.com/.default

### Contribute

- [x] Yes, I want to contribute

**Comment by bishalbera** [comment]
Ohh, so this was the issue 😅. I thought the value could vary, which is why I set it to be entered as user input. But now, after reading more carefully, I understand that we always use `https://graph.microsoft.com/.default` for the Microsoft Graph API. @morpulse

**Comment by morpulse** [comment]
Spot on @bishalbera :)

**Comment by bishalbera** [comment]
@morpulse ![Image](https://github.com/user-attachments/assets/512d7599-25ad-445b-b1a0-e08c8fba71db)
I had set that as a required field . Didn't you got an error when not passing it?

**Comment by morpulse** [comment]
I did pass it, but it didnt help I still got the 400 error since the scope was not set properly. after I set the scope properly it worked. the application_id_uri field is actually redundant as far as I can see.

**Comment by bishalbera** [comment]
okk . So do you want to raise the pr yourself or should  I do it? 

> I did pass it, but it didnt help I still got the 400 error since the scope was not set properly. after I set the scope properly it worked. the application_id_uri field is actually redundant as far as I can see.

**Comment by morpulse** [comment]
Sure, I'll raise it once I get contributor permission

**Comment by marcosmarxm** [comment]
@bishalbera can you submit the fix?

**Comment by bishalbera** [comment]
> [@bishalbera](https://github.com/bishalbera) can you submit the fix?

@marcosmarxm Yes sure. Also @morpulse just informed me about another bug regarding pagination which was working fine when we tested it. Never the less I fixed it again. I just had to change the stopping condition to `{{ response.get("@odata.nextLink") is none }}` from which i beilve was something like `is undefined`. So I'll fix both of these and raise a pr till tomorrow
![Image](https://github.com/user-attachments/assets/0f63b2f2-369b-4b21-9056-3f70b2c39ef8)
